src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -1841,5 +1855,202 @@ string[] linkingTargetFields
+                            ""object"": ""hybridSource"",
+                            ""type"":" + $"\"{sourceType}\"" + @"
+                        },
+                        ""permissions"": [
+                           {
+                            ""role"": ""anonymous"",
+                            ""actions"": [
+                               ""*""
+                            ]
+                           }
+                         ],
+                        ""rest"":{
+                            ""methods"":" + $"{methods}" + @"
+                         }
+                       }
+                    }
+                }";
+
+            RuntimeConfigLoader.TryParseConfig(runtimeConfigString, out RuntimeConfig runtimeConfig);
+
+            // Perform validation on the entity in the config and assert the expected results.
+            if (exceptionExpected)
+            {
+                DataApiBuilderException ex =
+                    Assert.ThrowsException<DataApiBuilderException>(() => RuntimeConfigValidator.ValidateEntityConfiguration(runtimeConfig));
+                Assert.AreEqual(expectedErrorMessage, ex.Message);
+                Assert.AreEqual(HttpStatusCode.ServiceUnavailable, ex.StatusCode);
+                Assert.AreEqual(DataApiBuilderException.SubStatusCodes.ConfigValidationError, ex.SubStatusCode);
+            }
+            else
+            {
+                RuntimeConfigValidator.ValidateEntityConfiguration(runtimeConfig);
+            }
+        }
+
+        /// <summary>
+        /// Test to validate that when multiple entities have the same custom rest path configured, we throw an exception.
+        /// </summary>
+        /// <param name="exceptionExpected">Whether an exception is expected as a result of test run.</param>
+        /// <param name="restPathForEntity">Custom rest path to be configured for the first entity.</param>
+        /// <param name="restPathForSecondEntity">Custom rest path to be configured for the second entity.</param>
+        /// <param name="expectedExceptionMessage">The expected exception message.</param>
+        [DataTestMethod]
+        [DataRow(true, "", "The rest path for entity: EntityA cannot be empty.",
+            DisplayName = "Empty rest path configured for an entity fails config validation.")]
+        [DataRow(false, "entityRestPath", DisplayName = "Rest path correctly configured as string.")]
+        public void ValidateRestPathForEntityInConfig(
+            bool exceptionExpected,
+            string restPathForEntity,
+            string expectedExceptionMessage = "")
+        {
+            Dictionary<string, Entity> entityMap = new();
+            Entity sampleEntity = GetSampleEntityUsingSourceAndRelationshipMap(
+                source: "TEST_SOURCEA",
+                relationshipMap: null,
+                graphQLDetails: null,
+                restDetails: new(new SupportedHttpVerb[] { }, restPathForEntity, true)
```

Comment:
Can we test if the entity name itself contains an invalid character and when that name becomes the restPath for the entity, would we give a validation error?

Reply:
Sure.

Reply:
Added